opm/material/fluidmatrixinteractions/EclHysteresisTwoPhaseLawParams.hpp

```diff
@@ -339,6 +338,15 @@ class EclHysteresisTwoPhaseLawParams : public EnsureFinalized
     Scalar Sncrt() const
     { return Sncrt_; }
 
+    Scalar SnTrapped() const
+    {
+        // For Killough the trapped saturation is already computed
+        if( config().krHysteresisModel() > 1 )
+            return Sncrt_;
```

Comment:
There are a couple of potential issues here.  First, the original code stored values in `Sncri_`, but this branch returns `Sncrt_`.  Is that intentional?  Secondly, the original code also would also consider `pcHysteresisModel() == 0`, but this exclusively uses `krHysteresisModel()`.  Does that mean we're missing something?

Reply:
sncri is the critical value of the relperm, sncrt is the trapped value from the hysteresis (which is the one we want). For Killough (krhysteresis > 1) this is computed directly, for Carlson only the shift is computed. Hence we need the sncri to get the trapped saturation for the Carlson model. That is also the reason that the sncri is now computed unconditionally regardless of hysteresis model. ( I removed the if above)